dissect/target/plugins/os/windows/network.py

```diff
@@ -0,0 +1,383 @@
+                        if isinstance(value, str):
+                            device_info["Vlan"] = PriorityVlanTagEnum.get(int(value), "Unknown")
+
+                if "NetCfgInstanceId" in device_info and "Type" in device_info and config:
+                    yield device_info
+
+    def _extract_network_device_config(self, interface_ids) -> list[dict, str]:
+        key = f"HKLM\\SYSTEM\\CurrentControlSet\\Services\\Tcpip\\Parameters\\Interfaces\\{interface_ids}"
+        fields = [
+            "DhcpDefaultGateway",
+            "DefaultGateway",
+            "DhcpIPAddress",
+            "IPAddress",
+            "DhcpNameServer",
+            "NameServer",
+            "DhcpSubnetMask",
+            "SubnetMask",
+            "DhcpDomain",
+            "Domain",
+            "InterfaceMetric",
+        ]
+        config = {"domain": [], "gateway": [], "ip": [], "dns": [], "subnetmask": [], "metric": []}
+
+        for field in fields:
+            try:
+                value = self.target.registry.key(key).value(field).value
+            except RegistryValueNotFoundError:
+                continue
+            except RegistryKeyNotFoundError:
+                break
+
+            if isinstance(value, str):
+                value = [value]
+            if isinstance(value, int):
+                value = [str(value)]
+
+            for i in value:
+                if field in ["DhcpDefaultGateway", "DefaultGateway"]:
+                    if i != "0.0.0.0":
+                        config["gateway"].append(i)
+
+                elif field in ["DhcpIPAddress", "IPAddress"]:
+                    config["ip"].append(i)
+
+                elif field in ["DhcpNameServer", "NameServer"]:
+                    if i != "":
+                        config["dns"].append(i)
+
+                elif field in ["SubnetMask", "DhcpSubnetMask"]:
+                    config["subnetmask"].append(i)
+
+                elif field in ["DhcpDomain", "Domain"]:
+                    if i != "":
+                        config["domain"].append(i)
+
+                elif field == "InterfaceMetric":
+                    if i != "":
+                        config["metric"].append(i)
+
+        if (
+            len(config["gateway"])
+            + len(config["ip"])
+            + len(config["dns"])
+            + len(config["subnetmask"])
+            + len(config["domain"])
+            + len(config["metric"])
+            > 0
+        ):
+            return config
+
+    def _extract_network_device_name(self, interface_ids) -> str:
+        key = (
+            "HKLM\\SYSTEM\\CurrentControlSet\\Control\\Network\\{4D36E972-E325-11CE-BFC1-08002BE10318}\\"
+            + f"{interface_ids}"
+            + "\\Connection"
+        )
+        try:
+            name = self.target.registry.key(key).value("Name").value
+            return name
+        except (RegistryValueNotFoundError, RegistryKeyNotFoundError):
+            return "Unknown"
+
+
+IfTypes = {
```

Comment:
Can you move this to the top of the file? And make it a proper `enum.Enum` maybe.